compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -280,6 +280,15 @@ impl<'tcx> ConstToPat<'tcx> {
                 slice: None,
                 suffix: Box::new([]),
             },
+            ty::Str => {
+                // String literal patterns may have type `str` if `deref_patterns` is enabled, in
+                // order to allow `deref!("..."): String`. Since we need a `&str` for the comparison
+                // when lowering to MIR in `Builder::perform_test`, treat the constant as a `&str`.
```

Comment:
I think it's important to note that the valtrees of `str` and `&str` are the same